base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/horizon_state_synchronization.rs

```diff
@@ -141,60 +158,71 @@ impl<'a, B: BlockchainBackend + 'static> HorizonStateSynchronization<'a, B> {
         let mut current_header = self.shared.db.fetch_header_containing_kernel_mmr(start + 1).await?;
         debug!(
             target: LOG_TARGET,
-            "Found current header in progress for kernels at mmr pos: {} height:{}",
+            "Found current header in progress for kernels at mmr pos: {} height: {}",
             start,
             current_header.height()
         );
+
         let mut kernels = vec![];
-        let mut txn = self.shared.db.write_transaction();
```

Comment:
This was pulled into the loop since the borrow checker won't allow the mutable reference for `set_state_info` while this immutable reference is held